### PR TITLE
Fix for #102

### PR DESCRIPTION
**Pull request by nderwin**
* added a test dependency on the Json implementation
* added a unit test to make sure the changes didn't break existing functionality, also adding test cases for null values
* changed the default returned value to a JsonValue of null, instead of a raw null

### CODE REVIEW

**Comment by nderwin** [comment]
Ooh, test failures I didn't see.  Closing this to fix.

**Comment by sberyozkin** [comment]
@nderwin Thanks for the PR, can you please type a sample JWT (in JSON format), I'd like to see which part of JWTCallerPrincipal throws NPE. The PR looks fine, but I'm not sure if both of your changes are needed or only one, difficult to appreciate if there could be some side-effects. 
Specifically I'm not sure that `JsonValue.NULL` has to be returned as a default value. There well may be a user code which already checks some claim values for `null`
Thanks

**Comment by nderwin** [comment]
Absolutely!  I also share your concern that there may be existing code expecting a Java null instead of a JSON null, so it may be necessary to limit the changes to values inside of collections.  An example JWT would be something like this:

```
{
  "sub": "some@body.com",
  "customClaim": {
    "customValue1": [
      "read"
    ],
    "customValue2": [
      null
    ],
    "customValue3": [
      "admin"
    ]
  },
  "exp": 1562957605
}
```

The trouble I was running into was with `customValue2` above - when the code would attempt to add the null to the JSON array build, it kicked it out with an exception (on purpose, looking at the source code for the implementation).

**Comment by sberyozkin** [comment]
@nderwin Thanks. I've debugged, and the only place in the code which IMHO needs an update is [this one](https://github.com/smallrye/smallrye-jwt/blob/master/implementation/src/main/java/io/smallrye/jwt/auth/principal/JWTCallerPrincipal.java#L193): add a `JsonValue.NULL`  if it is a null value.

Can you please update your PR ?

**Comment by nderwin** [comment]
Sure!  Just to be clear, we're talking about keeping the change in the `replaceMapClaims` method, and reverting the change in `wrapClaimValue`?

**Comment by sberyozkin** [comment]
@nderwin I believe the only change to address the issue you reported is needed inside the `wrapClaimValue` method at the point where the JsonArray is being prepared.

File: implementation/src/main/java/io/smallrye/jwt/auth/principal/JWTCallerPrincipal.java [hunk 1]
```diff
@@ -192,6 +194,8 @@ protected JsonValue wrapClaimValue(Object value) {
             for (Object element : list) {
                 if (element instanceof String) {
                     arrayBuilder.add(element.toString());
+                } else if (element == null) {
+                    arrayBuilder.add(NULL);
```
Comment (sberyozkin) [inline implementation/src/main/java/io/smallrye/jwt/auth/principal/JWTCallerPrincipal.java]: @nderwin thanks, this looks good, only a minor cosmetic issue: NULL is statically imported while just few lines above JsonValue.TRUE/FALSE are used. Lets make it consistent, I don't mind much if the static imports will be used, though have a minor preference for the `JsonValue` qualification in the code as it is easier to see what kind of NULL or Boolean values the code is dealing with, but you choose :-)

Comment (nderwin) [reply implementation/src/main/java/io/smallrye/jwt/auth/principal/JWTCallerPrincipal.java]: Ah, IDE auto-imports strike again.

**Comment by sberyozkin** [comment]
@nderwin thank you !
